.github/workflows/unit.yml

```diff
@@ -8,8 +8,13 @@ on:
   pull_request:
   workflow_dispatch:
   merge_group:
+  issue_comment:
+    types: [created]
 jobs:
   unit:
+    if: >-
+      github.event_name == 'pull_request' || 
+      (github.event_name == 'issue_comment' && startsWith(github.event.comment.body, '/retest unit'))
```

Comment:
I think this will mean we do not run unit tests on merge queue, etc.

Reply:
in the `on` section we have `on:
on-push-branches:- master`  that should trigger the workflow on commits pushed to master, including merges. But just to be sure, and more clear, we add other events and comments to the `on` section in f6b3a1e4

Reply:
I think `on` section was fine as it was originally. I would suggest reverting back.

My concern is within `if` which makes the `unit` job conditional. As it is right now - I think the workflow will be triggered on multiple events (push to `master`, pull request, merge queue, comment, etc), but the `unit` job will only be run on pull request events and comment events).

Or am I missing something?

Reply:
Ah, I think I get the problem you're finding. Yeah I think `on` can get reverted but `if` section needs to get modded to allow in all the other events we want unit to be run:
```yaml
jobs:
  unit:
    if: >-
      github.event_name == 'push' ||
      github.event_name == 'pull_request' ||
      github.event_name == 'workflow_dispatch' ||
      github.event_name == 'merge_group' ||
      (github.event_name == 'issue_comment' && startsWith(github.event.comment.body, '/retest unit'))
```
Seems right?

Reply:
Maybe something like `github.event_name != 'issue_comment' || startsWith(github.event.comment.body, '/retest unit')`?

So we don't have to update in two places if we change something in `on`?

Reply:
Yep, that's better! 39d866f cleans up the on section and adds that to if section